src/snowflake/snowpark/types.py

```diff
@@ -284,6 +284,23 @@ def _to_attributes(self) -> List["expression.Attribute"]:
     def __repr__(self) -> str:
         return f"StructType([{', '.join(repr(f) for f in self.fields)}])"
 
+    def __getitem__(self, item: Union[str, int, slice]) -> StructField:
+        """Access fields by name, index or slice."""
+        if isinstance(item, str):
+            for field in self.fields:
+                if field.name == item:
+                    return field
+            raise KeyError("No StructField named {0}".format(item))
```

Comment:
```suggestion
            raise KeyError(f"No StructField named {item}")
```

We prefer using f-strings in new code

Reply:
Changed